nixos/modules/services/misc/gitlab.nix

```diff
@@ -598,21 +591,33 @@ in {
 
         # JSON is a subset of YAML
         ln -fs ${pkgs.writeText "gitlab.yml" (builtins.toJSON gitlabConfig)} ${cfg.statePath}/config/gitlab.yml
-        ln -fs ${pkgs.writeText "database.yml" databaseYml} ${cfg.statePath}/config/database.yml
-        ln -fs ${pkgs.writeText "secrets.yml" secretsYml} ${cfg.statePath}/config/secrets.yml
+        cp -f  ${pkgs.writeText "database.yml" databaseYml} ${cfg.statePath}/config/database.yml
+        cp -f  ${pkgs.writeText "secrets.yml" secretsYml} ${cfg.statePath}/config/secrets.yml
         ln -fs ${pkgs.writeText "unicorn.rb" unicornConfig} ${cfg.statePath}/config/unicorn.rb
 
         chown -R ${cfg.user}:${cfg.group} ${cfg.statePath}/
         chmod -R ug+rwX,o-rwx+X ${cfg.statePath}/
 
+        DBPASSWORD=$(cat ${cfg.databasePassword})
+        SECRETKEY=$(cat ${cfg.secrets.secret})
+        OTPKEY=$(cat ${cfg.secrets.otp})
+        DBKEY=$(cat ${cfg.secrets.db})
+        JWSKEY=$(${pkgs.jq}/bin/jq -Rs . ${cfg.secrets.jws} | sed -e 's,\\n,\\\\n,g')
+
+        sed -e "s,#dbpasswordplaceholder#,$DBPASSWORD,g" -i ${cfg.statePath}/config/database.yml
+        sed -e "s,#secretkeyplaceholder#,$SECRETKEY,g"   -i ${cfg.statePath}/config/secrets.yml
+        sed -e "s,#otpkeyplaceholder#,$OTPKEY,g"         -i ${cfg.statePath}/config/secrets.yml
+        sed -e "s,#dbkeyplaceholder#,$DBKEY,g"           -i ${cfg.statePath}/config/secrets.yml
+        sed -e "s,#jwskeyplaceholder#,$JWSKEY,g"         -i ${cfg.statePath}/config/secrets.yml
```

Comment:
All these substitutions will break if any of the keys/passwords contain a comma, right? Perhaps use `${VAR//,/\\,/}` rather than `$VAR` in each case.

Reply:
If this seems like an acceptable fix to the substitution issue, I'll do it. I feel like the entire sed approach is wrong, but I don't know of any other reasonable alternatives. :/